### ISSUE TITLE
Fix code coverage of tests

### ISSUE THREAD

**Issue by Croydon** [nodegit/nodegit#1449]
Tests weren't run in the CI for some time. Therefore the failure of the tests were hidden.

I changed stuff in #1447 so they would get executed and disabled them temporary (in a visible, easy to switch way) due to the failure.

Eventually they should get fixed and enabled again.

https://travis-ci.org/inexorgame/nodegit/jobs/347256998#L9041

**Comment by rcjsuen** [comment]
> Tests weren't run in the CI for some time. Therefore the failure of the tests were hidden.

I don't understand what you mean, @Croydon. If I look at [nodegit/nodegit's build #3076](https://travis-ci.org/nodegit/nodegit/builds/346476440) for instance, the tests appear to have been run. You can scroll to the bottom of a [log file](https://api.travis-ci.org/v3/job/346476441/log.txt) and see output from Mocha in it.

**Comment by rcjsuen** [comment]
@Croydon Okay. I think what you mean is that the code coverage isn't being run, correct?

**Comment by JimiC** [comment]
Yes.And if someone can tell me what `before_script` I'll be able to sumbit my `travis` config proposal.

**Comment by Croydon** [comment]
No. It fails when it runs.

I have fixed it within my pull requests, that coverages tests get execited when you switch the variable. My last commit disabled coverage tests execution, so we can merge it now for better documentation.

No regression in testing between master and my pull request.
Improvements in documentation, building always for the latest Node minor versions and simplification in the .travis.yml file, visibility when and if coverages tests are getting executed.

Fixing tests are a follow up tasks, therefore this issue.

**Comment by JimiC** [comment]
@Croydon Notice https://github.com/rcjsuen as we discussed that it may be prudent to build docs only on release for the moment.

**Comment by rcjsuen** [comment]
I think the last time code coverage for the C++ files were run was in [Travis CI build #2741](https://travis-ci.org/nodegit/nodegit/builds/195995345). We can see mention of this at the bottom of its [log file](https://api.travis-ci.org/v3/job/195995347/log.txt).
```
...
Processing file src/commit.cc
Processing file src/revert.cc
Processing file src/odb.cc
Processing file src/transfer_progress.cc
Processing file src/signature.cc
Processing file src/proxy_options.cc
Processing file src/clone.cc
Processing file src/index_time.cc
Processing file src/cert_x509.cc
Processing file src/functions/sleep_for_ms.cc
Processing file src/functions/copy.cc
Processing file utils/execPromise.js
Writing directory view page.
Overall coverage rate:
  lines......: 39.6% (11417 of 28816 lines)
  functions..: no data found
  branches...: 71.9% (305 of 424 branches)

> [secure]@0.16.0 coveralls /home/travis/build/[secure]/[secure]
> cat ./test/coverage/merged.lcov | coveralls
```
This was then mistakenly "disabled" by 1269fd71b63c41f6f7ba22c1966efee17b1aaf12 because the coverage tests were then set to run if `$NODE_VERSION == "6"`. This condition never resolved to true because of `NODE_VERSION="6.5"`.

**Comment by Croydon** [comment]
Closing due to inactivity. If any active maintainer is interested in this, then please create a new issue.
